Framework/Algorithms/src/LineProfile.cpp

```diff
@@ -206,12 +243,18 @@ void profile(std::vector<double> &Xs, std::vector<double> &Ys,
     for (size_t j = limits.widthStart; j < limits.widthEnd; ++j) {
       const size_t iHor = dir == LineDirection::horizontal ? i : j;
       const size_t iVert = dir == LineDirection::horizontal ? j : i;
-      const double y = ws.y(iVert)[iHor];
+      double y = ws.y(iVert)[iHor];
       if ((ignoreNans && std::isnan(y)) || (ignoreInfs && std::isinf(y))) {
         continue;
       }
+      double e = ws.e(iVert)[iHor];
+      if (convertFromDistribution) {
+        const auto binWidth =
+            std::abs(ws.x(iVert)[iHor + 1] - ws.x(iVert)[iHor]);
+        y *= binWidth;
+        e *= binWidth;
```

Comment:
Have a look at `HistogramIterator` (or rather `HistogramItem` for the accessors). It provides thinks like bin width, might be simpler to use here.